### ISSUE TITLE
[Linux] Crash After Failed Thumbnail Download

### ISSUE THREAD

**Issue by ClaireSoftware** [libretro/RetroArch#10407]
m or a help section, this is strictly developer oriented

## Description

Segfault after failed thumbnail download

### Expected behavior

Thumbnails to download successfully or retroarch to fail gracefully

### Actual behavior

Retroarch segmentation faults.

### Steps to reproduce the bug

1. Download thumbnails using the included playlist thumbnail downloader
2. (maybe) fail the download
3. scroll through the xmb/ozone menu a bit

### Bisect Results

This bug has happened on multiple different versions of retroarch, including one i compiled myself.

### Version/Commit

- RetroArch: 0716fa02d64fafb12c399b973f0ed41b50fb267c

### Environment information

- OS: Arch Linux
- Compiler: GCC/Makefile

### log

https://pastebin.com/4nSBq0qz

**Comment by ClaireSoftware** [comment]
while trying to get the system information, I encountered a further issue where the task is set to 0x0. Perhaps that's part of the issue?

https://pastebin.com/LDc0miUb

**Comment by LibretroAdmin** [comment]
Might have been two years by now, but are you still able to reproduce this with the latest version of RetroArch (from source or nightlies)?

**Comment by KiriSweets** [comment]
> @ClaireSoftware If you don't reproduce the issue with latest stable, please close this issue.
> 
> You can make a test using [AppImage package](https://buildbot.libretro.com/stable/1.19.1/linux/x86_64/RetroArch.7z).
> 
> Thank you.

I've run into a rabbit hole of issues today that started with similar problems as the OP had. I downloaded the QT Appimage from https://www.retroarch.com/index.php?page=platforms

Started my day trying to get new cores because Sonic Rush was having issues with the DS core that's included with the Steam version. Ultimately decided to Switch to the non-Steam because I was having issues getting the new cores to read properly. Initially had a fine time with QT, but I pressed a whole bunch of "download thumbnails" "download metadata" etc type buttons and eventually while messing in other menus it crashed and wouldn't boot again.

That's how it started, but since then it's turned into a completely different issue (which may warrant its own topic but I'm posting here due to the shared initial problem). I tried completely resetting the config files but now whenever I try to change _any_ settings related to Fullscreen the app crashes and the cfg file becomes... Cursed? Like usually my changed settings don't save on a crash, but sometimes they will save only such that whatever caused the crash has been saved. I also can't find what lines changed in the cfg file to un-break the cfg. It should be noted that I've been experiencing these issues with no cores loaded.

So to recap in a less verbose fashion: I was downloading thumbnails, like OP. Went elsewhere in the menus to kill time, like OP. Eventually it crashed, like OP. I assume the thumbnails aren't actually related and what caused it was probably messing with Fullscreen options, because every time I touch those the cfg file acts like it's been corrupted and causes a crash and the cfg file to no longer work properly.

**Comment by LibretroAdmin** [comment]
Issue resolved
